oneTapClient no longer satisfies BetterAuthClientPlugin since 1.6.24 (TS2322 on getActions, TS2339 on oneTap)

## Summary

Since **1.6.24**, the `oneTapClient` plugin shipped by `better-auth/client/plugins` no longer type-checks against the `BetterAuthClientPlugin` type shipped by the same package. Passing it to `createAuthClient` fails with `TS2322`, and the resulting client loses the `oneTap` action (`TS2339`).

`1.6.23` is clean; `1.6.24` is the first bad version.

## Minimal reproduction

An 8-line project — no framework, no app config, `types: []` so nothing ambient leaks in.

`package.json`
```json
{ "name": "ba-repro", "private": true, "type": "module" }
```

`tsconfig.json`
```json
{ "compilerOptions": { "strict": true, "module": "esnext", "moduleResolution": "bundler",
  "target": "es2022", "jsx": "preserve", "skipLibCheck": true, "noEmit": true, "types": [] } }
```

`repro.ts`
```ts
import { createAuthClient } from "better-auth/react";
import { oneTapClient } from "better-auth/client/plugins";

const authClient = createAuthClient({
  plugins: [oneTapClient({ clientId: "x" })],
});

export const { oneTap } = authClient;
```

```
bun add better-auth@1.6.25 typescript@5.9.3 react@19.2.8 @types/react@19
bunx tsc --noEmit
```

## Bisect

Same project, only the `better-auth` version changed:

| version | `tsc --noEmit` |
|---------|----------------|
| 1.6.22  | pass |
| 1.6.23  | pass |
| **1.6.24** | **fail** |
| 1.6.25  | fail |

## Error

```
repro.ts(5,13): error TS2322: Type '{ id: "one-tap"; version: string; fetchPlugins: {...}[];
  getActions: ($fetch: BetterFetch<CreateFetchOption, unknown, unknown, Schema | undefined>,
  _: ClientStore) => {...}; getAtoms($fetch: BetterFetch<...>): {}; }'
  is not assignable to type 'BetterAuthClientPlugin'.
  Types of property 'getActions' are incompatible.
    Type '($fetch: BetterFetch<CreateFetchOption, unknown, unknown, Schema | undefined>,
      _: ClientStore) => { oneTap: (...) => Promise<...>; }'
    is not assignable to type '($fetch: BetterFetch, $store: ClientStore,
      options: BetterAuthClientOptions | undefined) => Record<string, any>'.
      Types of parameters '$fetch' and '$fetch' are incompatible.
        ...
        Type 'string | number | symbol' is not assignable to
          type '"params" | keyof PluginOptions | keyof O'.
```

In the real project this is followed by:

```
lib/auth-client.ts(15,45): error TS2339:
  Property 'oneTap' does not exist on type 'ReactAuthClient<BetterAuthClientOptions>'.
```

`oneTapClient`'s `getActions` is typed with the **generic** `BetterFetch<CreateFetchOption, unknown, unknown, Schema | undefined>`, while `BetterAuthClientPlugin.getActions` expects the **bare** `BetterFetch`. The variance falls out at `InferOptions` vs `PluginOptions` in the `$fetch` options parameter. Because the plugin fails to satisfy the interface, the client type never picks up its actions, hence the follow-on `TS2339`.

## Not a duplicate-dependency problem

Ruled this out before filing:

- `node_modules` contains exactly one copy of `@better-fetch/fetch` (`1.3.1`)
- `better-auth@1.6.25` pins `"@better-fetch/fetch": "1.3.1"` exactly
- `1.3.1` is the latest published version

So there is no dedupe, override or resolution that fixes it — the two types come from a single `@better-fetch/fetch` install.

## Impact

We're pinned to `^1.6.22` on a production app whose only login path is Google One Tap, so we're holding back auth patches. We'd rather not cast around a type error in the auth client, since that disables exactly the checking that would catch a real break.

## Environment

- better-auth 1.6.24 / 1.6.25 (1.6.22, 1.6.23 fine)
- typescript 5.9.3
- @better-fetch/fetch 1.3.1
- react 19.2.8
- bun 1.3.14, macOS

Happy to test a patch against the repro above.

--- a/.changeset/fix-one-tap-client-plugin-inference.md
+++ b/.changeset/fix-one-tap-client-plugin-inference.md
@@ -1,0 +1,5 @@
+---
+"better-auth": patch
+---
+
+Fix `oneTapClient()` collapsing `createAuthClient` type inference when combined with other client plugins. The `oneTap` action is available on the client again.

--- a/packages/better-auth/src/plugins/one-tap/client.ts
+++ b/packages/better-auth/src/plugins/one-tap/client.ts
@@ -1,9 +1,12 @@
 /// <reference types="@types/google.accounts" />
 import type {
+	BetterAuthClientOptions,
 	BetterAuthClientPlugin,
 	ClientFetchOption,
+	ClientStore,
 } from "@better-auth/core";
 import { isSafeUrlScheme } from "@better-auth/core/utils/url";
+import type { BetterFetch } from "@better-fetch/fetch";
 import { PACKAGE_VERSION } from "../../version";
 
 declare global {
@@ -212,7 +215,11 @@ export const oneTapClient = (options: GoogleOneTapOptions) => {
 				},
 			},
 		],
-		getActions: ($fetch, _) => {
+		getActions: (
+			$fetch: BetterFetch,
+			_$store: ClientStore,
+			_options: BetterAuthClientOptions | undefined,
+		) => {
 			return {
 				oneTap: async (
 					opts?: GoogleOneTapActionOptions | undefined,

--- a/packages/better-auth/src/plugins/one-tap/one-tap.test.ts
+++ b/packages/better-auth/src/plugins/one-tap/one-tap.test.ts
@@ -1,6 +1,9 @@
 // cspell:ignore AQAB
-import { afterEach, describe, expect, it, vi } from "vitest";
+import type { BetterAuthClientPlugin } from "@better-auth/core";
+import { afterEach, describe, expect, expectTypeOf, it, vi } from "vitest";
+import { createAuthClient } from "../../client";
 import { getTestInstance } from "../../test-utils/test-instance";
+import { oneTapClient } from "./client";
 import { oneTap } from "./index";
 
 vi.mock("@better-fetch/fetch", async (importOriginal) => {
@@ -710,5 +713,31 @@ describe("one-tap disableSignUp", () => {
 			status: "UNAUTHORIZED",
 			body: { message: "signup disabled" },
 		});
+	});
+});
+
+/**
+ * Declaration emit previously left `getActions`'s `$fetch` parameter
+ * un-annotated. With no import of `BetterFetch` in the file, emit had no
+ * name to reference and inlined `import("@better-fetch/fetch").BetterFetch`
+ * instead - a form not assignable to `BetterAuthClientPlugin`'s
+ * `getActions`, so `oneTapClient()` failed `createAuthClient`'s check and
+ * the client lost the `oneTap` action.
+ *
+ * @see https://github.com/better-auth/better-auth/issues/10583
+ */
+describe("oneTapClient types", () => {
+	it("should be assignable to BetterAuthClientPlugin", () => {
+		const plugin: BetterAuthClientPlugin = oneTapClient({
+			clientId: "test-client",
+		});
+		expect(plugin.id).toBe("one-tap");
+	});
+
+	it("should preserve the oneTap client action", () => {
+		const client = createAuthClient({
+			plugins: [oneTapClient({ clientId: "test-client" })],
+		});
+		expectTypeOf(client.oneTap).toBeFunction();
 	});
 });

